src/components/queryBuilder/views/LogsQueryBuilder.tsx

```diff
@@ -0,0 +1,310 @@
+import useIsNewQuery from 'hooks/useIsNewQuery';
+
+interface LogsQueryBuilderProps {
+  datasource: Datasource;
+  builderOptions: QueryBuilderOptions;
+  builderOptionsDispatch: React.Dispatch<BuilderOptionsReducerAction>;
+}
+
+interface LogsQueryBuilderState {
+  otelEnabled: boolean;
+  otelVersion: string;
+  selectedColumns: SelectedColumn[];
+  timeColumn?: SelectedColumn;
+  logLevelColumn?: SelectedColumn;
+  messageColumn?: SelectedColumn;
+  // liveView: boolean;
+  orderBy: OrderBy[];
+  limit: number;
+  filters: Filter[];
+}
+
+export const LogsQueryBuilder = (props: LogsQueryBuilderProps) => {
+  const { datasource, builderOptions, builderOptionsDispatch } = props;
+  const labels = allLabels.components.LogsQueryBuilder;
+  const allColumns = useColumns(datasource, builderOptions.database, builderOptions.table);
+  const isNewQuery = useIsNewQuery(builderOptions);
+  const builderState: LogsQueryBuilderState = useMemo(() => ({
+    otelEnabled: builderOptions.meta?.otelEnabled || false,
+    otelVersion: builderOptions.meta?.otelVersion || '',
+    timeColumn: getColumnByHint(builderOptions, ColumnHint.Time),
+    logLevelColumn: getColumnByHint(builderOptions, ColumnHint.LogLevel),
+    messageColumn: getColumnByHint(builderOptions, ColumnHint.LogMessage),
+    selectedColumns: builderOptions.columns?.filter(c => (
+      // Only select columns that don't have their own box
+      c.hint !== ColumnHint.Time &&
+      c.hint !== ColumnHint.LogLevel &&
+      c.hint !== ColumnHint.LogMessage
+    )) || [],
+    // liveView: builderOptions.meta?.liveView || false,
+    filters: builderOptions.filters || [],
+    orderBy: builderOptions.orderBy || [],
+    limit: builderOptions.limit || 0,
+    }), [builderOptions]);
+  const showConfigWarning = datasource.getDefaultLogsColumns().size === 0;
+
+  const onOptionChange = useBuilderOptionChanges<LogsQueryBuilderState>(next => {
+    const nextColumns = next.selectedColumns.slice();
+    if (next.timeColumn) {
+      nextColumns.push(next.timeColumn);
+    }
+    if (next.logLevelColumn) {
+      nextColumns.push(next.logLevelColumn);
+    }
+    if (next.messageColumn) {
+      nextColumns.push(next.messageColumn);
+    }
+
+    builderOptionsDispatch(setOptions({
+      columns: nextColumns,
+      filters: next.filters,
+      orderBy: next.orderBy,
+      limit: next.limit
+    }));
+  }, builderState);
+
+  useLogDefaultsOnMount(datasource, isNewQuery, builderOptions, builderOptionsDispatch);
+  useOtelColumns(builderState.otelEnabled, builderState.otelVersion, builderOptionsDispatch);
+  useDefaultTimeColumn(datasource, allColumns, builderOptions.table, builderState.timeColumn, builderState.otelEnabled, builderOptionsDispatch);
+  useDefaultFilters(builderOptions.table, builderState.timeColumn, builderState.filters, builderState.orderBy, builderOptionsDispatch);
+  
+  const configWarning = showConfigWarning && (
+    <Alert title="" severity="warning">
```

Comment:
can is be closed?

Reply:
Yes! Checked the docs again and they have `buttonContent` and `onRemove`. Added for Logs and Trace views.